Makefile

```diff
@@ -1,15 +1,26 @@
-# Build both steps required for executing the character count example
-all: processed_data/abyss.dat results/abyss.png results/results.txt
+SRCDIR := data
```

Comment:
I suggest a glossary of the used Make variables at the top of the file.  That would go a long way towards readability.

Reply:
Good idea - this is now done.

Reply:
What I meant was these variables:

- $%
- $@
- $<
- $(RM)

Maybe I'll make a edit with these